[Bug report] Unknown kotlin JVM target: 21 after adding photo manager

### Version

3.3.0

### Platforms

Android

### Device Model

pixel 8

### flutter info

```sh
[√] Flutter (Channel stable, 3.24.3, on Microsoft Windows [Version 10.0.26120.1930], locale en-US)
    • Flutter version 3.24.3 on channel stable at C:\dev\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 2663184aa7 (4 weeks ago), 2024-09-11 16:27:48 -0500
    • Engine revision 36335019a8
    • Dart version 3.5.3
    • DevTools version 2.37.3

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 35.0.0)
    • Android SDK at C:\Users\Alex\AppData\Local\Android\sdk
    • Platform android-35, build-tools 35.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 21.0.3+-12282718-b509.11)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[X] Visual Studio - develop Windows apps
    X Visual Studio not installed; this is necessary to develop Windows apps.
      Download at https://visualstudio.microsoft.com/downloads/.
      Please install the "Desktop development with C++" workload, including all of its default components

[√] Android Studio (version 2024.2)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 21.0.3+-12282718-b509.11)

[√] VS Code (version 1.94.1)
    • VS Code at C:\Users\Alex\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.98.0

[√] Connected device (4 available)
    • Pixel 7 (mobile)  • 29291FDH2003P7 • android-arm64  • Android 14 (API 34)
    • Windows (desktop) • windows        • windows-x64    • Microsoft Windows [Version 10.0.26120.1930]
    • Chrome (web)      • chrome         • web-javascript • Google Chrome 129.0.6668.101
    • Edge (web)        • edge           • web-javascript • Microsoft Edge 130.0.2849.27

[√] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```


### How to reproduce?

using android 35 with java 21 in the build options gives this error after adding photo_manager
compileOptions {
        sourceCompatibility = JavaVersion.VERSION_17
        targetCompatibility = JavaVersion.VERSION_17
    }

    kotlinOptions {
        jvmTarget = "17"
    }

### Logs

```sh
[        ] FAILURE: Build failed with an exception.
[   +1 ms] * What went wrong:
[        ] Execution failed for task ':photo_manager:compileDebugKotlin'.
[        ] > Error while evaluating property 'compilerOptions.jvmTarget' of task ':photo_manager:compileDebugKotlin'.
[        ]    > Failed to calculate the value of property 'jvmTarget'.
[        ]       > Unknown Kotlin JVM target: 21
[        ] * Try:
[        ] > Run with --debug option to get more log output.
[        ] > Run with --scan to get full insights.
[        ] > Get more help at https://help.gradle.org.
[   +1 ms] * Exception is:
[        ] org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':photo_manager:compileDebugKotlin'.
[        ]      at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:38)
[  +36 ms]      at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
[        ]      at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
[   +1 ms]      at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
[        ]      at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:209)
[        ]      at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
[        ]      at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
[   +1 ms]      at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
[        ]      at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:166)
[        ]      at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
[        ]      at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
[        ]      at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
[        ]      at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:42)
[        ]      at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:331)
[        ]      at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:318)
[        ]      at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.lambda$execute$0(DefaultTaskExecutionGraph.java:314)
[        ]      at org.gradle.internal.operations.CurrentBuildOperationRef.with(CurrentBuildOperationRef.java:85)
[        ]      at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:314)
[        ]      at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:303)
[        ]      at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:459)
[        ]      at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:376)
[        ]      at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
[        ]      at org.gradle.internal.concurrent.AbstractManagedExecutor$1.run(AbstractManagedExecutor.java:48)
[        ]      at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
[        ]      at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
[        ]      at java.base/java.lang.Thread.run(Unknown Source)
[        ] Caused by: org.gradle.api.internal.tasks.properties.PropertyEvaluationException: Error while evaluating property 'compilerOptions.jvmTarget' of task ':photo_manager:compileDebugKotlin'.
[        ]      at org.gradle.api.internal.tasks.properties.InputParameterUtils.prepareInputParameterValue(InputParameterUtils.java:32)
[        ]      at org.gradle.api.internal.tasks.execution.TaskExecution.lambda$visitRegularInputs$1(TaskExecution.java:317)
[        ]      at org.gradle.internal.execution.impl.DefaultInputFingerprinter$InputCollectingVisitor.visitInputProperty(DefaultInputFingerprinter.java:103)
[        ]      at org.gradle.api.internal.tasks.execution.TaskExecution.visitRegularInputs(TaskExecution.java:315)
[        ]      at org.gradle.internal.execution.impl.DefaultInputFingerprinter.fingerprintInputProperties(DefaultInputFingerprinter.java:63)
[        ]      at org.gradle.internal.execution.steps.AbstractCaptureStateBeforeExecutionStep.captureExecutionStateWithOutputs(AbstractCaptureStateBeforeExecutionStep.java:109)
[        ]      at org.gradle.internal.execution.steps.AbstractCaptureStateBeforeExecutionStep.lambda$captureExecutionState$0(AbstractCaptureStateBeforeExecutionStep.java:74)
[        ]      at org.gradle.internal.execution.steps.BuildOperationStep$1.call(BuildOperationStep.java:37)
[        ]      at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:209)
[        ]      at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
[        ]      at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
[        ]      at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
[        ]      at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:166)
[        ]      at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
[        ]      at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
[        ]      at org.gradle.internal.execution.steps.BuildOperationStep.operation(BuildOperationStep.java:34)
[        ]      at org.gradle.internal.execution.steps.AbstractCaptureStateBeforeExecutionStep.captureExecutionState(AbstractCaptureStateBeforeExecutionStep.java:69)
[        ]      at org.gradle.internal.execution.steps.AbstractCaptureStateBeforeExecutionStep.execute(AbstractCaptureStateBeforeExecutionStep.java:62)
[        ]      at org.gradle.internal.execution.steps.AbstractCaptureStateBeforeExecutionStep.execute(AbstractCaptureStateBeforeExecutionStep.java:43)
[        ]      at org.gradle.internal.execution.steps.AbstractSkipEmptyWorkStep.executeWithNonEmptySources(AbstractSkipEmptyWorkStep.java:125)
[        ]      at org.gradle.internal.execution.steps.AbstractSkipEmptyWorkStep.execute(AbstractSkipEmptyWorkStep.java:61)
[        ]      at org.gradle.internal.execution.steps.AbstractSkipEmptyWorkStep.execute(AbstractSkipEmptyWorkStep.java:36)
[        ]      at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsStartedStep.execute(MarkSnapshottingInputsStartedStep.java:38)
[        ]      at org.gradle.internal.execution.steps.LoadPreviousExecutionStateStep.execute(LoadPreviousExecutionStateStep.java:36)
[        ]      at org.gradle.internal.execution.steps.LoadPreviousExecutionStateStep.execute(LoadPreviousExecutionStateStep.java:23)
[        ]      at org.gradle.internal.execution.steps.HandleStaleOutputsStep.execute(HandleStaleOutputsStep.java:75)
[        ]      at org.gradle.internal.execution.steps.HandleStaleOutputsStep.execute(HandleStaleOutputsStep.java:41)
[        ]      at org.gradle.internal.execution.steps.AssignMutableWorkspaceStep.lambda$execute$0(AssignMutableWorkspaceStep.java:35)
[        ]      at org.gradle.api.internal.tasks.execution.TaskExecution$4.withWorkspace(TaskExecution.java:289)
[        ]      at org.gradle.internal.execution.steps.AssignMutableWorkspaceStep.execute(AssignMutableWorkspaceStep.java:31)
[        ]      at org.gradle.internal.execution.steps.AssignMutableWorkspaceStep.execute(AssignMutableWorkspaceStep.java:22)
[        ]      at org.gradle.internal.execution.steps.ChoosePipelineStep.execute(ChoosePipelineStep.java:40)
[        ]      at org.gradle.internal.execution.steps.ChoosePipelineStep.execute(ChoosePipelineStep.java:23)
[        ]      at org.gradle.internal.execution.steps.ExecuteWorkBuildOperationFiringStep.lambda$execute$2(ExecuteWorkBuildOperationFiringStep.java:67)
[        ]      at java.base/java.util.Optional.orElseGet(Unknown Source)
[        ]      at org.gradle.internal.execution.steps.ExecuteWorkBuildOperationFiringStep.execute(ExecuteWorkBuildOperationFiringStep.java:67)
[        ]      at org.gradle.internal.execution.steps.ExecuteWorkBuildOperationFiringStep.execute(ExecuteWorkBuildOperationFiringStep.java:39)
[        ]      at org.gradle.internal.execution.steps.IdentityCacheStep.execute(IdentityCacheStep.java:46)
[        ]      at org.gradle.internal.execution.steps.IdentityCacheStep.execute(IdentityCacheStep.java:34)
[        ]      at org.gradle.internal.execution.steps.IdentifyStep.execute(IdentifyStep.java:48)
[        ]      at org.gradle.internal.execution.steps.IdentifyStep.execute(IdentifyStep.java:35)
[        ]      at org.gradle.internal.execution.impl.DefaultExecutionEngine$1.execute(DefaultExecutionEngine.java:61)
[        ]      at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:127)
[        ]      at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:116)
[        ]      at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
[        ]      at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:51)
[        ]      at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
[        ]      at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:74)
[        ]      at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
[        ]      at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
[        ]      at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
[        ]      at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
[        ]      at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:209)
[        ]      at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
[        ]      at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
[        ]      at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
[        ]      at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:166)
[        ]      at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
[        ]      at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
[        ]      at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
[        ]      at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:42)
[        ]      at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:331)
[        ]      at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:318)
[        ]      at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.lambda$execute$0(DefaultTaskExecutionGraph.java:314)
[        ]      at org.gradle.internal.operations.CurrentBuildOperationRef.with(CurrentBuildOperationRef.java:85)
[        ]      at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:314)
[        ]      at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:303)
[        ]      at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:459)
[        ]      at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:376)
[        ]      at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
[        ]      at org.gradle.internal.concurrent.AbstractManagedExecutor$1.run(AbstractManagedExecutor.java:48)
[        ]      at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
[        ]      at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
[        ]      at java.base/java.lang.Thread.run(Unknown Source)
[        ] Caused by: org.gradle.api.internal.provider.AbstractProperty$PropertyQueryException: Failed to calculate the value of property 'jvmTarget'.
[        ]      at org.gradle.api.internal.provider.AbstractProperty.finalizeNow(AbstractProperty.java:284)
[        ]      at org.gradle.api.internal.provider.AbstractProperty.lambda$beforeRead$0(AbstractProperty.java:276)
[        ]      at org.gradle.api.internal.provider.ValueState.finalizeOnReadIfNeeded(ValueState.java:139)
[        ]      at org.gradle.api.internal.provider.AbstractProperty.beforeRead(AbstractProperty.java:276)
[        ]      at org.gradle.api.internal.provider.AbstractProperty.beforeRead(AbstractProperty.java:268)
[        ]      at org.gradle.api.internal.provider.AbstractProperty.calculateOwnValue(AbstractProperty.java:162)
[        ]      at org.gradle.api.internal.provider.AbstractMinimalProvider.getOrNull(AbstractMinimalProvider.java:105)
[        ]      at org.gradle.api.internal.provider.ProviderResolutionStrategy$1.resolve(ProviderResolutionStrategy.java:27)
[        ]      at org.gradle.util.internal.DeferredUtil.unpack(DeferredUtil.java:59)
[        ]      at org.gradle.util.internal.DeferredUtil.unpackOrNull(DeferredUtil.java:49)
[        ]      at org.gradle.api.internal.tasks.properties.InputParameterUtils.prepareInputParameterValue(InputParameterUtils.java:38)
[        ]      at org.gradle.api.internal.tasks.properties.InputParameterUtils.prepareInputParameterValue(InputParameterUtils.java:30)
[        ]      ... 73 more
[        ] Caused by: java.lang.IllegalArgumentException: Unknown Kotlin JVM target: 21
[        ]      at org.jetbrains.kotlin.gradle.dsl.JvmTarget$Companion.fromTarget(JvmTarget.kt:26)
[        ]      at org.jetbrains.kotlin.gradle.tasks.DefaultKotlinJavaToolchain$wireJvmTargetToToolchain$1$1.invoke(DefaultKotlinJavaToolchain.kt:78)
[        ]      at org.jetbrains.kotlin.gradle.tasks.DefaultKotlinJavaToolchain$wireJvmTargetToToolchain$1$1.invoke(DefaultKotlinJavaToolchain.kt:70)
[        ]      at org.jetbrains.kotlin.gradle.tasks.DefaultKotlinJavaToolchain$sam$org_gradle_api_Transformer$0.transform(DefaultKotlinJavaToolchain.kt)
[        ]      at org.gradle.api.internal.provider.ValueSupplier$Present.transform(ValueSupplier.java:518)
[        ]      at org.gradle.api.internal.provider.TransformBackedProvider.mapValue(TransformBackedProvider.java:91)
[        ]      at org.gradle.api.internal.provider.TransformBackedProvider.calculateOwnValue(TransformBackedProvider.java:82)
[        ]      at org.gradle.api.internal.provider.AbstractMinimalProvider.calculateValue(AbstractMinimalProvider.java:115)
[        ]      at org.gradle.api.internal.provider.TransformBackedProvider.calculateOwnValue(TransformBackedProvider.java:81)
[        ]      at org.gradle.api.internal.provider.AbstractMinimalProvider.calculateValue(AbstractMinimalProvider.java:115)
[        ]      at org.gradle.api.internal.provider.AbstractMinimalProvider.withFinalValue(AbstractMinimalProvider.java:168)
[        ]      at org.gradle.api.internal.provider.DefaultProperty.finalValue(DefaultProperty.java:162)
[        ]      at org.gradle.api.internal.provider.DefaultProperty.finalValue(DefaultProperty.java:33)
[        ]      at org.gradle.api.internal.provider.AbstractProperty.finalizeNow(AbstractProperty.java:281)
[        ]      ... 84 more
[        ] BUILD FAILED in 7s
```


### Example code (optional)

_No response_

### Contact

_No response_

--- a/.github/workflows/build-example-apk.yml
+++ b/.github/workflows/build-example-apk.yml
@@ -7,7 +7,7 @@ on:
 
 env:
   GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-  NDK_VERSION: r27.0.12077973
+  NDK_VERSION: r27
 
 jobs:
   build-apk:

--- a/.github/workflows/build-example-apk.yml
+++ b/.github/workflows/build-example-apk.yml
@@ -1,5 +1,9 @@
 name: Build example apk
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
 on:
   release:
     types:

--- a/.github/workflows/build-example-apk.yml
+++ b/.github/workflows/build-example-apk.yml
@@ -10,7 +10,6 @@ permissions:
 
 env:
   GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-  NDK_VERSION: r27
 
 jobs:
   build-apk:
@@ -21,10 +20,6 @@ jobs:
         with:
           distribution: 'zulu'
           java-version: '17'
-      - uses: nttld/setup-ndk@v1
-        with:
-          ndk-version: ${{ env.NDK_VERSION }}
-          link-to-sdk: true
       - uses: flutter-actions/setup-flutter@v4
         with:
           channel: 'stable'

--- a/.github/workflows/build-example-apk.yml
+++ b/.github/workflows/build-example-apk.yml
@@ -5,6 +5,9 @@ on:
     types:
       - created
 
+permissions:
+  contents: write
+
 env:
   GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
   NDK_VERSION: r27
@@ -16,13 +19,13 @@ jobs:
       - uses: actions/checkout@v4
       - uses: actions/setup-java@v4
         with:
-          distribution: 'adopt'
+          distribution: 'zulu'
           java-version: '17'
       - uses: nttld/setup-ndk@v1
         with:
           ndk-version: ${{ env.NDK_VERSION }}
           link-to-sdk: true
-      - uses: subosito/flutter-action@v2
+      - uses: flutter-actions/setup-flutter@v4
         with:
           channel: 'stable'
           cache: true
@@ -37,6 +40,3 @@ jobs:
           echo "Show apk download url: "
           gh release view ${{ github.event.release.tag_name }} --json assets --jq '.assets.[].url'
         working-directory: ${{ github.workspace }}/example
-
-permissions:
-  contents: write

--- a/.github/workflows/build-example-apk.yml
+++ b/.github/workflows/build-example-apk.yml
@@ -7,6 +7,7 @@ on:
 
 env:
   GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  NDK_VERSION: r27.0.12077973
 
 jobs:
   build-apk:
@@ -17,6 +18,10 @@ jobs:
         with:
           distribution: 'adopt'
           java-version: '17'
+      - uses: nttld/setup-ndk@v1
+        with:
+          ndk-version: ${{ env.NDK_VERSION }}
+          link-to-sdk: true
       - uses: subosito/flutter-action@v2
         with:
           channel: 'stable'

--- a/.github/workflows/check-compatibility.yml
+++ b/.github/workflows/check-compatibility.yml
@@ -18,18 +18,19 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
-      - uses: subosito/flutter-action@v2
+      - uses: flutter-actions/setup-flutter@v4
         with:
           channel: 'stable'
           cache: true
       - uses: actions/setup-java@v4
         with:
-          distribution: 'adopt'
+          distribution: 'zulu'
           java-version: '17'
       - uses: nttld/setup-ndk@v1
         with:
           ndk-version: ${{ env.NDK_VERSION }}
           link-to-sdk: true
+      - run: sudo echo "y" | sudo $ANDROID_HOME/tools/bin/sdkmanager "ndk;27.0.12077973"
       - run: flutter doctor -v
         name: Flutter info
       - run: flutter create new_project --platforms=android
@@ -48,7 +49,7 @@ jobs:
     runs-on: macos-latest
     steps:
       - uses: actions/checkout@v4
-      - uses: subosito/flutter-action@v2
+      - uses: flutter-actions/setup-flutter@v4
         with:
           channel: 'stable'
           cache: true
@@ -79,7 +80,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
-      - uses: subosito/flutter-action@v2
+      - uses: flutter-actions/setup-flutter@v4
         with:
           channel: 'stable'
           cache: true
@@ -105,7 +106,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
-      - uses: subosito/flutter-action@v2
+      - uses: flutter-actions/setup-flutter@v4
         with:
           channel: 'stable'
           cache: true
@@ -137,7 +138,7 @@ jobs:
     runs-on: macos-latest
     steps:
       - uses: actions/checkout@v4
-      - uses: subosito/flutter-action@v2
+      - uses: flutter-actions/setup-flutter@v4
         with:
           channel: 'stable'
           cache: true
@@ -168,7 +169,7 @@ jobs:
     runs-on: windows-latest
     steps:
       - uses: actions/checkout@v4
-      - uses: subosito/flutter-action@v2
+      - uses: flutter-actions/setup-flutter@v4
         with:
           channel: 'stable'
           cache: true

--- a/.github/workflows/check-compatibility.yml
+++ b/.github/workflows/check-compatibility.yml
@@ -31,7 +31,7 @@ jobs:
         uses: android-actions/setup-android@v3
       - name: Install NDK
         run: |
-          echo "sdkmanager --install 'ndk;27.0.12077973'" | sdkmanager --stdin
+          sudo echo "y" | sudo sdkmanager "ndk;27.0.12077973"
       - run: flutter doctor -v
         name: Flutter info
       - run: flutter create new_project --platforms=android

--- a/.github/workflows/check-compatibility.yml
+++ b/.github/workflows/check-compatibility.yml
@@ -1,5 +1,9 @@
 name: Add library to new flutter project
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
 on:
   pull_request:
     branches:

--- a/.github/workflows/check-compatibility.yml
+++ b/.github/workflows/check-compatibility.yml
@@ -29,9 +29,6 @@ jobs:
           java-version: '17'
       - name: Set up Android SDK
         uses: android-actions/setup-android@v3
-      - name: Install NDK
-        run: |
-          echo "y" | sdkmanager --install "ndk;27.0.12077973"
       - run: flutter doctor -v
         name: Flutter info
       - run: flutter create new_project --platforms=android

--- a/.github/workflows/check-compatibility.yml
+++ b/.github/workflows/check-compatibility.yml
@@ -9,9 +9,6 @@ on:
       - main
       - 2.x
 
-env:
-  NDK_VERSION: r27
-
 jobs:
   build-for-android:
     name: Build for Android on ubuntu-latest
@@ -26,11 +23,11 @@ jobs:
         with:
           distribution: 'zulu'
           java-version: '17'
-      - uses: nttld/setup-ndk@v1
-        with:
-          ndk-version: ${{ env.NDK_VERSION }}
-          link-to-sdk: true
-      - run: sudo echo "y" | sudo $ANDROID_HOME/tools/bin/sdkmanager "ndk;27.0.12077973"
+      - name: Set up Android SDK
+        uses: android-actions/setup-android@v3
+      - name: Install NDK
+        run: |
+          echo "sdkmanager --install 'ndk;27.0.12077973'" | sdkmanager --stdin
       - run: flutter doctor -v
         name: Flutter info
       - run: flutter create new_project --platforms=android
@@ -53,10 +50,6 @@ jobs:
         with:
           channel: 'stable'
           cache: true
-      - uses: nttld/setup-ndk@v1
-        with:
-          ndk-version: ${{ env.NDK_VERSION }}
-          link-to-sdk: true
       - run: flutter doctor -v
         name: Flutter info
       - run: flutter create new_project --platforms=ios
@@ -84,10 +77,6 @@ jobs:
         with:
           channel: 'stable'
           cache: true
-      - uses: nttld/setup-ndk@v1
-        with:
-          ndk-version: ${{ env.NDK_VERSION }}
-          link-to-sdk: true
       - run: flutter doctor -v
         name: Flutter info
       - run: flutter create new_project --platforms=web
@@ -114,10 +103,6 @@ jobs:
         run: |
               sudo apt-get update -y
               sudo apt-get install -y ninja-build libgtk-3-dev
-      - uses: nttld/setup-ndk@v1
-        with:
-          ndk-version: ${{ env.NDK_VERSION }}
-          link-to-sdk: true
       - name: Enable support for linux
         run: flutter config --enable-linux-desktop
       - run: flutter doctor -v
@@ -142,10 +127,6 @@ jobs:
         with:
           channel: 'stable'
           cache: true
-      - uses: nttld/setup-ndk@v1
-        with:
-          ndk-version: ${{ env.NDK_VERSION }}
-          link-to-sdk: true
       - run: flutter doctor -v
         name: Flutter info
       - run: flutter create new_project --platforms=macos
@@ -173,10 +154,6 @@ jobs:
         with:
           channel: 'stable'
           cache: true
-      - uses: nttld/setup-ndk@v1
-        with:
-          ndk-version: ${{ env.NDK_VERSION }}
-          link-to-sdk: true
       - run: flutter doctor -v
         name: Flutter info
       - run: flutter create new_project --platforms=windows

--- a/.github/workflows/check-compatibility.yml
+++ b/.github/workflows/check-compatibility.yml
@@ -31,7 +31,7 @@ jobs:
         uses: android-actions/setup-android@v3
       - name: Install NDK
         run: |
-          sudo echo "y" | sudo sdkmanager "ndk;27.0.12077973"
+          echo "y" | sdkmanager --install "ndk;27.0.12077973"
       - run: flutter doctor -v
         name: Flutter info
       - run: flutter create new_project --platforms=android

--- a/.github/workflows/check-compatibility.yml
+++ b/.github/workflows/check-compatibility.yml
@@ -2,12 +2,15 @@ name: Add library to new flutter project
 
 on:
   pull_request:
-    branches: 
+    branches:
       - main
   push:
-    branches: 
+    branches:
       - main
       - 2.x
+
+env:
+  NDK_VERSION: r27.0.12077973
 
 jobs:
   build-for-android:
@@ -23,6 +26,10 @@ jobs:
         with:
           distribution: 'adopt'
           java-version: '17'
+      - uses: nttld/setup-ndk@v1
+        with:
+          ndk-version: ${{ env.NDK_VERSION }}
+          link-to-sdk: true
       - run: flutter doctor -v
         name: Flutter info
       - run: flutter create new_project --platforms=android
@@ -45,6 +52,10 @@ jobs:
         with:
           channel: 'stable'
           cache: true
+      - uses: nttld/setup-ndk@v1
+        with:
+          ndk-version: ${{ env.NDK_VERSION }}
+          link-to-sdk: true
       - run: flutter doctor -v
         name: Flutter info
       - run: flutter create new_project --platforms=ios
@@ -72,6 +83,10 @@ jobs:
         with:
           channel: 'stable'
           cache: true
+      - uses: nttld/setup-ndk@v1
+        with:
+          ndk-version: ${{ env.NDK_VERSION }}
+          link-to-sdk: true
       - run: flutter doctor -v
         name: Flutter info
       - run: flutter create new_project --platforms=web
@@ -98,6 +113,10 @@ jobs:
         run: |
               sudo apt-get update -y
               sudo apt-get install -y ninja-build libgtk-3-dev
+      - uses: nttld/setup-ndk@v1
+        with:
+          ndk-version: ${{ env.NDK_VERSION }}
+          link-to-sdk: true
       - name: Enable support for linux
         run: flutter config --enable-linux-desktop
       - run: flutter doctor -v
@@ -122,6 +141,10 @@ jobs:
         with:
           channel: 'stable'
           cache: true
+      - uses: nttld/setup-ndk@v1
+        with:
+          ndk-version: ${{ env.NDK_VERSION }}
+          link-to-sdk: true
       - run: flutter doctor -v
         name: Flutter info
       - run: flutter create new_project --platforms=macos
@@ -149,6 +172,10 @@ jobs:
         with:
           channel: 'stable'
           cache: true
+      - uses: nttld/setup-ndk@v1
+        with:
+          ndk-version: ${{ env.NDK_VERSION }}
+          link-to-sdk: true
       - run: flutter doctor -v
         name: Flutter info
       - run: flutter create new_project --platforms=windows

--- a/.github/workflows/check-compatibility.yml
+++ b/.github/workflows/check-compatibility.yml
@@ -10,7 +10,7 @@ on:
       - 2.x
 
 env:
-  NDK_VERSION: r27.0.12077973
+  NDK_VERSION: r27
 
 jobs:
   build-for-android:

--- a/.github/workflows/runnable.yml
+++ b/.github/workflows/runnable.yml
@@ -13,7 +13,7 @@ on:
       - "**.md"
 
 env:
-  NDK_VERSION: r27.0.12077973
+  NDK_VERSION: r27
 
 jobs:
   analyze:

--- a/.github/workflows/runnable.yml
+++ b/.github/workflows/runnable.yml
@@ -12,9 +12,6 @@ on:
     paths-ignore:
       - "**.md"
 
-env:
-  NDK_VERSION: r27
-
 jobs:
   analyze:
     name: Analyze on ${{ matrix.os }}
@@ -98,10 +95,6 @@ jobs:
         with:
           distribution: 'zulu'
           java-version: '17'
-      - uses: nttld/setup-ndk@v1
-        with:
-          ndk-version: ${{ env.NDK_VERSION }}
-          link-to-sdk: true
       - uses: subosito/flutter-action@v2
         with:
           channel: "stable"
@@ -144,10 +137,6 @@ jobs:
         with:
           distribution: 'zulu'
           java-version: '17'
-      - uses: nttld/setup-ndk@v1
-        with:
-          ndk-version: ${{ env.NDK_VERSION }}
-          link-to-sdk: true
       - uses: subosito/flutter-action@v2
         with:
           channel: "stable"
@@ -169,10 +158,6 @@ jobs:
         with:
           distribution: 'zulu'
           java-version: '17'
-      - uses: nttld/setup-ndk@v1
-        with:
-          ndk-version: ${{ env.NDK_VERSION }}
-          link-to-sdk: true
       - uses: subosito/flutter-action@v2
         with:
           channel: "stable"
@@ -194,10 +179,6 @@ jobs:
         with:
           distribution: 'zulu'
           java-version: '17'
-      - uses: nttld/setup-ndk@v1
-        with:
-          ndk-version: ${{ env.NDK_VERSION }}
-          link-to-sdk: true
       - uses: flutter-actions/setup-flutter@v4
         with:
           channel: 'stable'

--- a/.github/workflows/runnable.yml
+++ b/.github/workflows/runnable.yml
@@ -26,8 +26,8 @@ jobs:
       - uses: actions/checkout@v4
       - uses: actions/setup-java@v4
         with:
-          distribution: "adopt"
-          java-version: "17"
+          distribution: 'zulu'
+          java-version: '17'
       - uses: subosito/flutter-action@v2
         with:
           channel: "stable"
@@ -60,8 +60,8 @@ jobs:
       - uses: actions/checkout@v4
       - uses: actions/setup-java@v4
         with:
-          distribution: "adopt"
-          java-version: "17"
+          distribution: 'zulu'
+          java-version: '17'
       - uses: subosito/flutter-action@v2
         with:
           channel: "stable"
@@ -96,8 +96,8 @@ jobs:
       - uses: actions/checkout@v4
       - uses: actions/setup-java@v4
         with:
-          java-version: "17"
-          distribution: "adopt"
+          distribution: 'zulu'
+          java-version: '17'
       - uses: nttld/setup-ndk@v1
         with:
           ndk-version: ${{ env.NDK_VERSION }}
@@ -142,8 +142,8 @@ jobs:
       - uses: actions/checkout@v4
       - uses: actions/setup-java@v4
         with:
-          java-version: "17"
-          distribution: "adopt"
+          distribution: 'zulu'
+          java-version: '17'
       - uses: nttld/setup-ndk@v1
         with:
           ndk-version: ${{ env.NDK_VERSION }}
@@ -167,8 +167,8 @@ jobs:
       - uses: actions/checkout@v4
       - uses: actions/setup-java@v4
         with:
-          java-version: "17"
-          distribution: "adopt"
+          distribution: 'zulu'
+          java-version: '17'
       - uses: nttld/setup-ndk@v1
         with:
           ndk-version: ${{ env.NDK_VERSION }}
@@ -192,16 +192,15 @@ jobs:
       - uses: actions/checkout@v4
       - uses: actions/setup-java@v4
         with:
-          distribution: "adopt"
-          java-version: "17"
+          distribution: 'zulu'
+          java-version: '17'
       - uses: nttld/setup-ndk@v1
         with:
           ndk-version: ${{ env.NDK_VERSION }}
           link-to-sdk: true
-      - uses: subosito/flutter-action@v2
+      - uses: flutter-actions/setup-flutter@v4
         with:
-          channel: "stable"
-          architecture: x64
+          channel: 'stable'
           cache: true
       - run: flutter doctor -v
       - run: flutter pub get

--- a/.github/workflows/runnable.yml
+++ b/.github/workflows/runnable.yml
@@ -1,5 +1,9 @@
 name: Runnable (stable)
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
 on:
   push:
     branches:

--- a/.github/workflows/runnable.yml
+++ b/.github/workflows/runnable.yml
@@ -12,6 +12,9 @@ on:
     paths-ignore:
       - "**.md"
 
+env:
+  NDK_VERSION: r27.0.12077973
+
 jobs:
   analyze:
     name: Analyze on ${{ matrix.os }}
@@ -95,6 +98,10 @@ jobs:
         with:
           java-version: "17"
           distribution: "adopt"
+      - uses: nttld/setup-ndk@v1
+        with:
+          ndk-version: ${{ env.NDK_VERSION }}
+          link-to-sdk: true
       - uses: subosito/flutter-action@v2
         with:
           channel: "stable"
@@ -137,6 +144,10 @@ jobs:
         with:
           java-version: "17"
           distribution: "adopt"
+      - uses: nttld/setup-ndk@v1
+        with:
+          ndk-version: ${{ env.NDK_VERSION }}
+          link-to-sdk: true
       - uses: subosito/flutter-action@v2
         with:
           channel: "stable"
@@ -158,6 +169,10 @@ jobs:
         with:
           java-version: "17"
           distribution: "adopt"
+      - uses: nttld/setup-ndk@v1
+        with:
+          ndk-version: ${{ env.NDK_VERSION }}
+          link-to-sdk: true
       - uses: subosito/flutter-action@v2
         with:
           channel: "stable"
@@ -179,6 +194,10 @@ jobs:
         with:
           distribution: "adopt"
           java-version: "17"
+      - uses: nttld/setup-ndk@v1
+        with:
+          ndk-version: ${{ env.NDK_VERSION }}
+          link-to-sdk: true
       - uses: subosito/flutter-action@v2
         with:
           channel: "stable"

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -1,5 +1,3 @@
-import java.util.regex.Matcher
-import java.util.regex.Pattern
 
 group 'com.fluttercandies.photo_manager'
 version '1.0-SNAPSHOT'
@@ -26,96 +24,6 @@ rootProject.allprojects {
 apply plugin: 'com.android.library'
 apply plugin: 'kotlin-android'
 
-void log(Object msg) {
-    logger.log(LogLevel.INFO, "[PhotoManager] $msg")
-}
-
-/**
- * Get java version from JAVA_HOME
- * @param path JAVA_HOME path
- * @return java version
- */
-JavaVersion getJavaFromHome(String path) {
-    def releaseFile = new File(path, "release")
-
-    if (releaseFile.exists()) {
-        def pattern = Pattern.compile("JAVA_VERSION=\"(\\d+)\"")
-        def matcher = pattern.matcher(releaseFile.text)
-        if (matcher.find()) {
-            def version = matcher.group(1)
-            log("Get java version from release file: $version")
-            return JavaVersion.toVersion(version)
-        }
-    } else {
-        def javaBin = new File("${path}/bin/java")
-        if (javaBin.exists()) {
-            return JavaVersion.VERSION_1_8
-        } else if (new File("${path}/bin/java.exe").exists()) {
-            return JavaVersion.VERSION_1_8
-        }
-
-        return null
-    }
-}
-
-JavaVersion getJavaVersion() {
-    try {
-        // 0. Get java.version property from project's gradle.properties
-        // The version value is like 1.8, 11, 17, 21, etc.
-        // Also see: https://docs.gradle.org/current/javadoc/org/gradle/api/JavaVersion.html#toVersion-java.lang.Object-
-        JavaVersion res
-
-        def javaVersion = project.rootProject.findProperty("java.version")
-        if (javaVersion != null) {
-            res = JavaVersion.toVersion(javaVersion)
-            log("Get java version from project's gradle.properties: $javaVersion")
-            if (res != null) {
-                return res
-            }
-        }
-
-        String javaHome
-        // 1. read from JAVA_HOME environment variable
-        javaHome = System.getenv("JAVA_HOME")
-        if (res == null && javaHome != null) {
-            res = getJavaFromHome(javaHome)
-            log("Get java version from JAVA_HOME: $javaHome")
-            if (res != null) {
-                return res
-            }
-        }
-
-        // 2. read gradle.properties
-        javaHome = project.rootProject.findProperty("java.home")
-        if (res == null && javaHome != null) {
-            res = getJavaFromHome(javaHome)
-            log("Get java version from gradle.properties: $javaHome")
-            if (res != null) {
-                return res
-            }
-        }
-
-        // 3. read from property with java.home
-        javaHome = System.getProperty("java.home")
-        if (res == null && javaHome != null) {
-            res = getJavaFromHome(javaHome)
-            log("Get java version from java.home: $javaHome")
-            if (res != null) {
-                return res
-            }
-        }
-
-        if (res != null) {
-            return res
-        }
-    } catch (Exception e) {
-        e.printStackTrace()
-    }
-
-    // last, use default version with current
-    return JavaVersion.current()
-}
-
 android {
     if (project.android.hasProperty('namespace') ||
             getGradle().getGradleVersion().substring(0, 1).toInteger() >= 8) {
@@ -124,19 +32,12 @@ android {
 
     compileSdkVersion 34
 
-    sourceSets {
-        main.java.srcDirs += 'src/main/kotlin'
-    }
-
-    compileOptions {
-        def version = getJavaVersion()
-        sourceCompatibility = version
-        targetCompatibility = version
-    }
-
     defaultConfig {
         minSdkVersion 16
-        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
+    }
+
+    sourceSets {
+        main.java.srcDirs += 'src/main/kotlin'
     }
 }
 

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -35,6 +35,11 @@ android {
         minSdkVersion 16
     }
 
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_17
+        targetCompatibility JavaVersion.VERSION_17
+    }
+
     sourceSets {
         main.java.srcDirs += 'src/main/kotlin'
     }

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -25,8 +25,7 @@ apply plugin: 'com.android.library'
 apply plugin: 'kotlin-android'
 
 android {
-    if (project.android.hasProperty('namespace') ||
-            getGradle().getGradleVersion().substring(0, 1).toInteger() >= 8) {
+    if (project.android.hasProperty('namespace')) {
         namespace 'com.flutterandies.photo_manager'
     }
 

--- a/android/gradle.properties
+++ b/android/gradle.properties
@@ -1,2 +1,0 @@
-org.gradle.jvmargs=-Xmx1536M
-kotlin.code.style=official

--- a/android/gradle/wrapper/gradle-wrapper.properties
+++ b/android/gradle/wrapper/gradle-wrapper.properties
@@ -1,6 +1,0 @@
-#Mon Sep 16 14:30:05 CST 2019
-distributionBase=GRADLE_USER_HOME
-distributionPath=wrapper/dists
-zipStoreBase=GRADLE_USER_HOME
-zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.5.1-all.zip

--- a/example/android/app/build.gradle
+++ b/example/android/app/build.gradle
@@ -17,14 +17,10 @@ android {
     namespace 'com.fluttercandies.photo_manager_example'
 
     compileSdk flutter.compileSdkVersion
+    ndkVersion = "27.0.12077973"
 
     kotlinOptions {
         jvmTarget = '17'
-    }
-
-    compileOptions {
-        sourceCompatibility JavaVersion.current()
-        targetCompatibility JavaVersion.current()
     }
 
     sourceSets {

--- a/example/android/app/build.gradle
+++ b/example/android/app/build.gradle
@@ -23,6 +23,11 @@ android {
         jvmTarget = '17'
     }
 
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_17
+        targetCompatibility JavaVersion.VERSION_17
+    }
+
     sourceSets {
         main.java.srcDirs += 'src/main/kotlin'
     }

--- a/example/android/app/build.gradle
+++ b/example/android/app/build.gradle
@@ -26,7 +26,6 @@ if (flutterVersionName == null) {
 android {
     namespace 'com.fluttercandies.photo_manager_example'
 
-    compileSdk 35
 
     signingConfigs {
         release {
@@ -36,6 +35,7 @@ android {
             keyPassword '123456'
         }
     }
+    compileSdk flutter.compileSdkVersion
 
     kotlinOptions {
         jvmTarget = '17'
@@ -57,6 +57,7 @@ android {
         versionCode flutterVersionCode.toInteger()
         versionName flutterVersionName
         multiDexEnabled true
+        targetSdk = flutter.targetSdkVersion
     }
 
     buildTypes {

--- a/example/android/app/build.gradle
+++ b/example/android/app/build.gradle
@@ -13,28 +13,9 @@ if (localPropertiesFile.exists()) {
     }
 }
 
-def flutterVersionCode = localProperties.getProperty('flutter.versionCode')
-if (flutterVersionCode == null) {
-    flutterVersionCode = '1'
-}
-
-def flutterVersionName = localProperties.getProperty('flutter.versionName')
-if (flutterVersionName == null) {
-    flutterVersionName = '1.0'
-}
-
 android {
     namespace 'com.fluttercandies.photo_manager_example'
 
-
-    signingConfigs {
-        release {
-            storeFile file("${rootDir.absolutePath}/photo-manager-keystore")
-            storePassword '123456'
-            keyAlias 'key0'
-            keyPassword '123456'
-        }
-    }
     compileSdk flutter.compileSdkVersion
 
     kotlinOptions {
@@ -50,29 +31,33 @@ android {
         main.java.srcDirs += 'src/main/kotlin'
     }
 
-
     defaultConfig {
         applicationId "com.fluttercandies.photo_manager_example"
         minSdkVersion 21
-        versionCode flutterVersionCode.toInteger()
-        versionName flutterVersionName
-        multiDexEnabled true
         targetSdk = flutter.targetSdkVersion
+        versionCode = flutter.versionCode
+        versionName = flutter.versionName
+    }
+
+    signingConfigs {
+        forAll {
+            storeFile file("${rootDir.absolutePath}/photo-manager-keystore")
+            storePassword '123456'
+            keyAlias 'key0'
+            keyPassword '123456'
+        }
     }
 
     buildTypes {
         debug {
-            signingConfig signingConfigs.release
+            signingConfig signingConfigs.forAll
         }
         profile {
-            signingConfig signingConfigs.release
+            signingConfig signingConfigs.forAll
         }
         release {
-            signingConfig signingConfigs.release
+            signingConfig signingConfigs.forAll
         }
-    }
-    lint {
-        disable 'InvalidPackage'
     }
 }
 

--- a/example/android/build.gradle
+++ b/example/android/build.gradle
@@ -3,39 +3,36 @@ allprojects {
         google()
         mavenCentral()
     }
-    subprojects {
-        afterEvaluate { project ->
-            if (project.plugins.hasPlugin("com.android.application") ||
-                    project.plugins.hasPlugin("com.android.library")) {
-                project.android {
-                    compileSdkVersion 35
-                    buildToolsVersion "35.0.0"
-                    if (namespace == null) {
-                        namespace project.group
-                    }
+}
 
-                    ndkVersion = "27.0.12077973"
+rootProject.buildDir = '../build'
+subprojects {
+    project.buildDir = "${rootProject.buildDir}/${project.name}"
+    afterEvaluate { project ->
+        if (project.plugins.hasPlugin("com.android.application") ||
+                project.plugins.hasPlugin("com.android.library")) {
+            project.android {
+                compileSdkVersion 35
+                buildToolsVersion "35.0.0"
+                if (namespace == null) {
+                    namespace project.group
+                }
 
-                    compileOptions {
-                        sourceCompatibility JavaVersion.VERSION_17
-                        targetCompatibility JavaVersion.VERSION_17
-                    }
+                ndkVersion = "27.0.12077973"
 
-                    if(project.android.hasProperty("kotlinOptions")) {
-                        kotlinOptions {
-                            jvmTarget = '17'
-                        }
+                compileOptions {
+                    sourceCompatibility JavaVersion.VERSION_17
+                    targetCompatibility JavaVersion.VERSION_17
+                }
+
+                if (project.android.hasProperty("kotlinOptions")) {
+                    kotlinOptions {
+                        jvmTarget = '17'
                     }
                 }
             }
         }
     }
-}
-
-
-rootProject.buildDir = '../build'
-subprojects {
-    project.buildDir = "${rootProject.buildDir}/${project.name}"
 }
 subprojects {
     project.evaluationDependsOn(':app')

--- a/example/android/build.gradle
+++ b/example/android/build.gradle
@@ -8,31 +8,6 @@ allprojects {
 rootProject.buildDir = '../build'
 subprojects {
     project.buildDir = "${rootProject.buildDir}/${project.name}"
-    afterEvaluate { project ->
-        if (project.plugins.hasPlugin("com.android.application") ||
-                project.plugins.hasPlugin("com.android.library")) {
-            project.android {
-                compileSdkVersion 35
-                buildToolsVersion "35.0.0"
-                if (namespace == null) {
-                    namespace project.group
-                }
-
-                ndkVersion = "27.0.12077973"
-
-                compileOptions {
-                    sourceCompatibility JavaVersion.VERSION_17
-                    targetCompatibility JavaVersion.VERSION_17
-                }
-
-                if (project.android.hasProperty("kotlinOptions")) {
-                    kotlinOptions {
-                        jvmTarget = '17'
-                    }
-                }
-            }
-        }
-    }
 }
 subprojects {
     project.evaluationDependsOn(':app')

--- a/example/android/gradle.properties
+++ b/example/android/gradle.properties
@@ -1,4 +1,4 @@
-org.gradle.jvmargs=-Xmx1536M
+org.gradle.jvmargs=-Xmx4G -XX:MaxMetaspaceSize=2G -XX:+HeapDumpOnOutOfMemoryError
 android.useAndroidX=true
 android.enableJetifier=true
 kotlin.code.style=official

--- a/example/android/settings.gradle
+++ b/example/android/settings.gradle
@@ -18,8 +18,8 @@ pluginManagement {
 
 plugins {
     id "dev.flutter.flutter-plugin-loader" version "1.0.0"
-    id("com.android.application") version "8.9.1" apply false
-    id("com.android.library") version "8.9.1" apply false
+    id "com.android.application" version "8.9.1" apply false
+    id "com.android.library" version "8.9.1" apply false
     id "org.jetbrains.kotlin.android" version "2.1.20" apply false
     id "org.jetbrains.kotlin.kapt" version "2.1.20" apply false
 }

